### PR TITLE
work in progress

### PR DESCRIPTION
**Pull request by jo-pol**
@janvanmansum 

Trying to add something to an existing folder now no longer complains about not finding it, but has the following incorrect content in cfg.json (and probably more problems):

```
"relations":[{
  "predicate":"http://dans.knaw.nl/ontologies/relations#isMemberOf",
  "objectSDO":"easy-dataset_3"
}
```

the third line should be something like

```
"object":"info:fedora/easy-folder:999"
```

### CODE REVIEW

File: src/main/scala/nl/knaw/dans/easy/stage/fileitem/EasyFilesAndFolders.scala [hunk 1]
```diff
@@ -1,18 +1,3 @@
-/**
```
Comment (janvanmansum) [inline src/main/scala/nl/knaw/dans/easy/stage/fileitem/EasyFilesAndFolders.scala:1]: @jo-pol please merge the comment header. Or add it with `mvn license:format`

File: src/main/scala/nl/knaw/dans/easy/stage/fileitem/EasyFilesAndFolders.scala [hunk 2]
```diff
@@ -48,16 +33,19 @@ object EasyFilesAndFolders {
...
-          val resultSet = query.executeQuery()
-          if(!resultSet.next()) throw new RuntimeException("Count query returned no rows (?) A count query should ALWAYS return one row")
-          resultSet.getString("count") == "1"
-        })
+      .map(_.replaceAll("/$", ""))
+      .filter(!_.isEmpty)
+    val parentPath = subPaths.find(path => {
+        query.setString(1, path)
+        val resultSet = query.executeQuery()
+        // TODO don't forget to close the query?
```
Comment (janvanmansum) [inline src/main/scala/nl/knaw/dans/easy/stage/fileitem/EasyFilesAndFolders.scala]: @jo-pol Isn't it closed below on line 49?

File: src/main/scala/nl/knaw/dans/easy/stage/fileitem/EasyFilesAndFolders.scala [hunk 3]
```diff
@@ -48,16 +33,19 @@ object EasyFilesAndFolders {
 
   def getExistingParent(pathInDataset: String, datasetId: String): Try[Option[String]] = Try {
     val query: PreparedStatement = conn.prepareStatement("SELECT count(pid) FROM easy_folders WHERE path = ?")
-    val parentPath = pathInDataset
+    val subPaths = pathInDataset
```
Comment (janvanmansum) [inline src/main/scala/nl/knaw/dans/easy/stage/fileitem/EasyFilesAndFolders.scala]: @jo-pol: What are sub-paths?

File: src/main/scala/nl/knaw/dans/easy/stage/fileitem/EasyFilesAndFolders.scala (hunk 2 above)
Comment (jo-pol) [inline src/main/scala/nl/knaw/dans/easy/stage/fileitem/EasyFilesAndFolders.scala]: Not when you throw, but as the message says: it should not occur. Besides that it is a pity you loose the id here and have to fetch it later again. Neither have much priority at the moment.

File: src/main/scala/nl/knaw/dans/easy/stage/fileitem/EasyFilesAndFolders.scala (hunk 3 above)
Comment (jo-pol) [inline src/main/scala/nl/knaw/dans/easy/stage/fileitem/EasyFilesAndFolders.scala]: I suppose you remember what you have been doing here. Would something like parentPaths be a better name?

File: src/main/scala/nl/knaw/dans/easy/stage/fileitem/EasyFilesAndFolders.scala (hunk 2 above)
Comment (janvanmansum) [inline src/main/scala/nl/knaw/dans/easy/stage/fileitem/EasyFilesAndFolders.scala]: Point taken. We should probably make sure it is always closed, `executeQuery` or `resultSet.getString` might also throw an exception.

File: src/main/scala/nl/knaw/dans/easy/stage/fileitem/EasyFilesAndFolders.scala (hunk 3 above)
Comment (janvanmansum) [inline src/main/scala/nl/knaw/dans/easy/stage/fileitem/EasyFilesAndFolders.scala]: The intent was to get the first existing ancestor. So, maybe the function should be called `getExistingAncestor` and the variable `ancestors` (if you insist on using a temp for this).

Comment (jo-pol) [inline src/main/scala/nl/knaw/dans/easy/stage/fileitem/EasyFilesAndFolders.scala]: I had a hard time reading what happens here (the first time I saw a scanleft), a temp can clarify a lot and makes debugging easier.

Comment (janvanmansum) [inline src/main/scala/nl/knaw/dans/easy/stage/fileitem/EasyFilesAndFolders.scala]: No problem, but then rename the function and variable as suggested above.

**Comment by janvanmansum** [comment]
Changes incorporated in https://github.com/DANS-KNAW/easy-stage-dataset/pull/55

**Pull request closed without merging**